code/core/src/common/js-package-manager/NPMProxy.ts

```diff
@@ -89,13 +89,11 @@ export class NPMProxy extends JsPackageManager {
     packageName: string,
     basePath = this.cwd
   ): Promise<PackageJson | null> {
-    const packageJsonPath = await findUp(
-      (dir) => {
-        const possiblePath = join(dir, 'node_modules', packageName, 'package.json');
-        return existsSync(possiblePath) ? possiblePath : undefined;
-      },
-      { cwd: basePath }
-    );
+    const dirs = walk.up(basePath ?? '.');
```

Comment:
same as in this comment:
https://github.com/storybookjs/storybook/pull/31338/files#r2075527459